infrastructure/lib/addons/perfdash.ts

```diff
@@ -0,0 +1,241 @@
+import { Construct } from 'constructs';
+import { Aws, CfnStack, StackProps } from 'aws-cdk-lib';
+import { aws_eks as eks } from 'aws-cdk-lib';
+import { aws_iam as iam } from 'aws-cdk-lib';
+
+export interface PerfDashProps extends StackProps {
+  cluster: eks.Cluster;
+  namespace: string;
+}
+
+export class PerfDash extends Construct {
+  constructor(scope: Construct, id: string, props: PerfDashProps) {
+    super(scope, id);
+
+    const ns = props.cluster.addManifest('perfdash-namespace', {
+      apiVersion: 'v1',
+      kind: 'Namespace',
+      metadata: {
+          name: props.namespace
+      }
+  })
+
+    const sa = props.cluster.addServiceAccount('perfdash-sa', {
+      name: 'perfdash-log-fetcher',
+      namespace: props.namespace
+    });
+
+    sa.node.addDependency(ns)
+    sa.role.attachInlinePolicy(new iam.Policy(this, 'perfdash-policy', {
+      statements: [
+        new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              // S3 readonly access
+              "s3:Get*",
+              "s3:List*",
+              "s3-object-lambda:Get*",
+              "s3-object-lambda:List*",
+            ],
+        }),
+      ],
+  }));
+
+  const cm = props.cluster.addManifest('perfdash-cm', {
+    apiVersion: 'v1',
+    kind: 'ConfigMap',
+    metadata: {
+        name: "perfdash-config",
+        namespace: props.namespace,
+        labels: {
+            "app.kubernetes.io/name": "perfdash",
+            "app.kubernetes.io/component": "config",
+            "app.kubernetes.io/part-of": "perfdash",
+        },
+    },
+    data: {
+        "PERFDASH_LOG_BUCKET": "my-bucket-name",
+        "AWS_DEFAULT_REGION": "us-west-2",
+        "AWS_REGION": "us-west-2",
+        "AWS_ROLE_ARN": sa.role.roleArn,
+        "jobs.yaml": `periodics:
+- name: "kit-eks-1k"
+  tags:
+  - "perfDashPrefix: 1k test"
+  - "perfDashJobType: performance"
+  - "perfDashArtifactsDir: "
+- name: "kit-eks-2k"
+  tags:
+  - "perfDashPrefix: 2k test"
+  - "perfDashJobType: performance"
+  - "perfDashArtifactsDir: "
+  `
+    },
+  })
+  cm.node.addDependency(ns)
+
+  const deploy = props.cluster.addManifest('perfdash-deploy', {
```

Comment:
So far theme of the code is, we install addons that have dependency with AWS resources with CDK and addons without aws resource dependencies through just flux. 
What's your intent in cdkfying this over the current way which is being applied through flux ?

Reply:
As I mentioned in the PR description, new approach create an IAM Role with proper permissions and bind it to the k8s serviceaccount while the old approach expect user to create the Role with proper permission and then manually edit the k8s service account to enable the binding.

Reply:
I wasn't referring to the IRSA part here, just the deployment spec part of it. I'm fine with the IRSA part of it.

Reply:
Like for example today we install tekton tests policy sa account through CDK here https://github.com/awslabs/kubernetes-iteration-toolkit/blob/main/infrastructure/lib/kit-infrastructure.ts#L141-L162 but won't rest of the spec through flux.

Reply:
Or other way is to install through CDK but install through helm or yaml reference instead of having the spec completely embedded into the cdk code like for example here https://github.com/awslabs/kubernetes-iteration-toolkit/blob/main/infrastructure/lib/addons/karpenter.ts#L63-L88

Reply:
> Like for example today we install tekton tests policy sa account through CDK here https://github.com/awslabs/kubernetes-iteration-toolkit/blob/main/infrastructure/lib/kit-infrastructure.ts#L141-L162 but rest of the spec through flux.

When we move CM out, we have resources: namespace, serviceaccount, deployment, service in CDK.
SA depends on NS. Deployment depends on NS and SA, since it needs `sa.role.roleArn`. Svc seems to be the only resource can be moved out of CDK. But I don't feel it make much sense to do it for just one resource.

> Or other way is to install through CDK but install through helm or yaml reference instead of having the spec completely embedded into the cdk code like for example here https://github.com/awslabs/kubernetes-iteration-toolkit/blob/main/infrastructure/lib/addons/karpenter.ts#L63-L88

PerfDash is not distributed through helm. If we want to do it, we need to create and maintain a helm chart for it. IMO we may not want to maintain a chart when we don't want to get into the distribution business for PerfDash.

@hakuna-matatah WDYT?